qtpy/cli.py

```diff
@@ -0,0 +1,81 @@
+import argparse
+import inspect
+
+
+class RawDescriptionArgumentDefaultsHelpFormatter(
+    argparse.RawDescriptionHelpFormatter,
+    argparse.ArgumentDefaultsHelpFormatter,
+):
+    pass
+
+
+def cli():
+    """Features in support of development with QtPy."""
+
+    parser = argparse.ArgumentParser(
+        description=inspect.getdoc(cli),
+        formatter_class=RawDescriptionArgumentDefaultsHelpFormatter,
+    )
+
+    parser.set_defaults(func=parser.print_help)
+
+    cli_subparsers = parser.add_subparsers()
+
+    mypy_parser = cli_subparsers.add_parser(
+        name=mypy.__name__,
+        description=inspect.getdoc(mypy),
+        formatter_class=RawDescriptionArgumentDefaultsHelpFormatter,
+    )
+    mypy_parser.set_defaults(func=mypy)
+
+    mypy_subparsers = mypy_parser.add_subparsers()
+
+    mypy_args_parser = mypy_subparsers.add_parser(
+        name=args.__name__,
+        description=inspect.getdoc(args),
+        formatter_class=RawDescriptionArgumentDefaultsHelpFormatter,
+    )
+    mypy_args_parser.set_defaults(func=args)
+
+    arguments = parser.parse_args()
+
+    reserved_parameters = {'func'}
+    cleaned = {
+        k: v
+        for k, v in vars(args).items()
+        if k not in reserved_parameters
+    }
+
+    arguments.func(**cleaned)
+
+
+def mypy():
+    """Features in support of running mypy"""
+
+    pass
+
+
+def args():
+    """Generate command line arguments for using mypy with QtPy.
+
+    This will generate strings similar to the following which help guide mypy
+    through which library QtPy would have used so that mypy can get the proper
+    underlying type hints.
+
+        --always-false=PYQT4 --always-false=PYQT5 --always-false=PYSIDE --always-true=PYSIDE2
+
+    Use such as:
+
+        env/bin/mypy --package mypackage $(env/bin/qtpy mypy args)
+    """
+
+    options = {False: '--always-false', True: '--always-true'}
+
+    import qtpy
+
+    apis_active = {name: qtpy.API.lower() == name.lower() for name in qtpy.APIS}
+    print(' '.join(
+        f'{options[is_active]}={name.upper()}'
+        for name, is_active
+        in apis_active.items()
+    ))
```

Comment:
```suggestion
    ))


def main():
    cli()


if __name__ == "__main__":
    main()
```

A bit of boilerplate to make it a little easier to run as a script in case the user doesn't actually have QtPy installed, but rather just checked out locally

Reply:
Good points, agreed this isn't necessary or desirable.

> And why the indirection through main()?

Just convention, and probably a foolish one, at least in this case where it doesn't do anything special.

> Would want a sys.exit()?

Not sure the need for `sys.exit()`, when it would exist anyway with `0` upon completion?